src/main/java/lc/misc/DPISetter.java

```diff
@@ -0,0 +1,69 @@
+package lc.misc;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Iterator;
+
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageTypeSpecifier;
+import javax.imageio.ImageWriteParam;
+import javax.imageio.ImageWriter;
+import javax.imageio.metadata.IIOInvalidTreeException;
+import javax.imageio.metadata.IIOMetadata;
+import javax.imageio.metadata.IIOMetadataNode;
+import javax.imageio.stream.ImageOutputStream;
+import java.awt.image.BufferedImage;
+
+public class DPISetter {
```

Comment:
Looks good. Some comments about design and clarity:

* If specify the functions to be `static`, there will be no need to instantiate the class.
* Rename the class to `PngImageWriter`, and function `setDPI` to `write`.